src/gwt/src/org/rstudio/studio/client/workbench/views/tutorial/TutorialPane.java

```diff
@@ -409,6 +447,74 @@ private final native void initTutorialJsCallbacks()
       
    }-*/;
    
+   private final native void initExternalWindowJsCallbacks(WindowEx window,
+                                                           String tutorialUrl,
+                                                           String tutorialName,
+                                                           String tutorialPackage)
+   /*-{
+      
+      // register this window
+      $wnd.tutorialWindows = $wnd.tutorialWindows || {};
+      $wnd.tutorialWindows[tutorialUrl] = {
+         "package": tutorialPackage,
+         "name": tutorialName,
+         "window": window
+      };
+      
+      // start polling for window closure
```

Comment:
This feels pretty ugly, but I couldn't find a better way to get notified of a closed child window that worked both in server and desktop sessions.

Reply:
If you can get it working in server mode, there's a cheat for the desktop: have Qt tell you when a window is closed. This is how we handle closure of source windows.

https://github.com/rstudio/rstudio/blob/9679854e04ab6b1358e4798250d535ef82b9f9e3/src/cpp/desktop/DesktopSatelliteWindow.cpp#L81